v1/plugins/logs/eventBuffer.go

```diff
@@ -0,0 +1,259 @@
+package logs
+
+import (
+	"bytes"
+	"compress/gzip"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"sync"
+
+	"github.com/open-policy-agent/opa/v1/plugins/rest"
+)
+
+// eventBuffer stores and uploads a gzip compressed JSON array of EventV1 entries
+type eventBuffer struct {
+	buffer               chan []byte    // buffer stores JSON encoded EventV1 data
+	mtx                  sync.RWMutex   // mtx controls access to the buffer
+	payload              *payloadBuffer // payload contains the compressed JSON array to be uploaded
+	client               rest.Client    // client is used to upload the data to the configured service
+	uploadPath           string         // uploadPath is the configured HTTP resource path for upload
+	uploadSizeLimitBytes int64          // uploadSizeLimitBytes will enforce a maximum payload size to be uploaded
+}
+
+func newEventBuffer(bufferSizeLimitEvents int64, client rest.Client, uploadPath string, uploadSizeLimitBytes int64) *eventBuffer {
+	return &eventBuffer{
+		buffer:               make(chan []byte, bufferSizeLimitEvents),
+		payload:              newPayloadBuffer(),
+		client:               client,
+		uploadPath:           uploadPath,
+		uploadSizeLimitBytes: uploadSizeLimitBytes,
+	}
+}
+
+// Reconfigure updates the user configurable values
+func (b *eventBuffer) Reconfigure(bufferSizeLimitEvents int64, client rest.Client, uploadPath string, uploadSizeLimitBytes int64) (int, []error) {
+	b.mtx.Lock()
+	defer b.mtx.Unlock()
+
+	b.client = client
+	b.uploadPath = uploadPath
+	oldUploadSizeLimitBytes := b.uploadSizeLimitBytes
+	b.uploadSizeLimitBytes = uploadSizeLimitBytes
+
+	if int64(cap(b.buffer)) == bufferSizeLimitEvents && uploadSizeLimitBytes == oldUploadSizeLimitBytes {
+		return 0, nil
+	}
+
+	close(b.buffer)
+	newBuffer := make(chan []byte, bufferSizeLimitEvents)
```

Comment:
If `int64(cap(b.buffer)) == bufferSizeLimitEvents`, maybe we don't need to recreate the channel and flush it's content?

Reply:
With the serialization now only done on the consumer side this became simpler so it only recreates a channel if the limit is changed.